Scripts/Utilities/MaterialUtils.cs

```diff
@@ -105,6 +105,14 @@ public static Color RandomColor()
 			return new Color(r, g, b);
 		}
 
+		public static Color HueShift(Color color, float shift)
```

Comment:
That still won't work because color is a struct - I was just assuming it would also return the color that was shifted, but I see that Vector3's don't do things that way so yes, your current method (non extension is best)